From 0e91911d22d43621c820d7f5b28be671050fd290 Mon Sep 17 00:00:00 2001
From: Aki Tuomi <aki.tuomi@open-xchange.com>
Date: Mon, 27 May 2019 09:43:25 +0300
Subject: [PATCH] doveadm-sieve: Fix script synchronization

When dsyncing, this codepath is always called with prefix "".
There is no point checking the prefix at all.

Broken in 479c5e57046dec76078597df844daccbfc0eb75f
---
 src/plugins/doveadm-sieve/doveadm-sieve-sync.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/plugins/doveadm-sieve/doveadm-sieve-sync.c b/src/plugins/doveadm-sieve/doveadm-sieve-sync.c
index 34bd18d4..069f20ab 100644
--- a/src/plugins/doveadm-sieve/doveadm-sieve-sync.c
+++ b/src/plugins/doveadm-sieve/doveadm-sieve-sync.c
@@ -606,9 +606,7 @@ sieve_attribute_iter_init(struct mailbox *box, enum mail_attribute_type type,
        siter->iter.box = box;
        siter->super = sbox->super.attribute_iter_init(box, type, prefix);

-       if (box->storage->user->dsyncing &&
-           type == MAIL_ATTRIBUTE_TYPE_PRIVATE &&
-           str_begins(prefix, MAILBOX_ATTRIBUTE_PREFIX_SIEVE)) {
+       if (box->storage->user->dsyncing) {
                if (sieve_attribute_iter_script_init(siter) < 0)
                        siter->failed = TRUE;
        }
